bc_obps/registration/tests/endpoints/test_operations_endpoints.py

```diff
@@ -0,0 +1,468 @@
+        assert post_response.status_code == 401
+        post_response = post_response = TestUtils.mock_post_with_auth_role(
+            self, "cas_analyst", content_type_json, mock_operation.json(), endpoint=None
+        )
+        assert post_response.status_code == 401
+        post_response = post_response = TestUtils.mock_post_with_auth_role(
+            self, "cas_pending", content_type_json, mock_operation.json()
+        )
+        assert post_response.status_code == 401
+
+    def test_unauthorized_roles_cannot_put_operations(self, client):
+
+        operation = baker.make(Operation)
+        mock_operation = TestUtils.mock_OperationUpdateIn()
+        # IRC users can't put
+        response = TestUtils.mock_put_with_auth_role(
+            self,
+            'cas_pending',
+            content_type_json,
+            mock_operation.json(),
+            self.endpoint + "/" + str(operation.id) + "?submit=false",
+        )
+        assert response.status_code == 401
+        response = TestUtils.mock_put_with_auth_role(
+            self,
+            'cas_admin',
+            content_type_json,
+            mock_operation.json(),
+            self.endpoint + "/" + str(operation.id) + "?submit=false",
+        )
+        assert response.status_code == 401
+        response = TestUtils.mock_put_with_auth_role(
+            self,
+            'cas_analyst',
+            content_type_json,
+            mock_operation.json(),
+            self.endpoint + "/" + str(operation.id) + "?submit=false",
+        )
+        assert response.status_code == 401
+        # unapproved industry users cannot put
+        response = TestUtils.mock_put_with_auth_role(
+            self,
+            'industry_user',
+            content_type_json,
+            mock_operation.json(),
+            self.endpoint + "/" + str(operation.id) + "?submit=false",
+        )
+        assert response.status_code == 401
+        response = TestUtils.mock_put_with_auth_role(
+            self,
+            'industry_user_admin',
+            content_type_json,
+            mock_operation.json(),
+            self.endpoint + "/" + str(operation.id) + "?submit=false",
+        )
+        assert response.status_code == 401
+
+    def test_unauthorized_users_cannot_update_status(self):
+        operation = baker.make(Operation)
+
+        url = self.build_update_status_url(operation_id=operation.id)
+
+        response = TestUtils.mock_put_with_auth_role(
+            self, "industry_user", content_type_json, {"status": "approved"}, url
+        )
+        assert response.status_code == 401
+        response = TestUtils.mock_put_with_auth_role(
+            self, "industry_user_admin", content_type_json, {"status": "approved"}, url
+        )
+        assert response.status_code == 401
+
+    def test_get_method_for_200_status(self, client):
+        # IRC users can get all operations
```

Comment:
Brianna, note to self, coordinate this one with https://app.zenhub.com/workspaces/team-moose-sprint-board-60d216175a2689000ec774d0/issues/gh/bcgov/cas-registration/421